frontend/packages/dev-console/src/components/health-checks/AddHealthChecks.tsx

```diff
@@ -117,7 +120,7 @@ const AddHealthChecks: React.FC<FormikProps<FormikValues> & AddHealthChecksProps
             errorMessage={status && status?.errors?.json?.message}
             isSubmitting={isSubmitting}
             submitLabel={healthCheckAdded ? 'Save' : 'Add'}
-            disableSubmit={!dirty || !_.isEmpty(errors)}
+            disableSubmit={isFormClean || !_.isEmpty(errors)}
```

Comment:
I am not able to remove the probes as the Save button gets disabled when I remove the probes. 
See GIF
![health-check-validation](https://user-images.githubusercontent.com/2561818/82558101-9a9d1780-9b8a-11ea-9449-2ec8f0f41984.gif)

Reply:
Updated. PTAL

Reply:
Any reason to remove the check for ```dirty```?

Reply:
The form modifies the `dirty` prop even if the probe section is simply opened, without modifying any of the fields.

Reply:
I would suggest keeping the dirty check as well. 
```dirty``` returns ```true``` if values are not deeply equal from initial values, ```false``` otherwise
```suggestion
            disableSubmit={isFormClean || !dirty || !_.isEmpty(errors)}
```

Reply:
Added.